examples/3d/3d_custom_image.rs

```diff
@@ -0,0 +1,186 @@
+            [1.0, 1.0, 0.0],
+            // back    `  `  (-x)
+            [0.0, 0.0, 0.0],
+            [0.0, 0.0, 1.0],
+            [0.0, 1.0, 1.0],
+            [0.0, 1.0, 0.0],
+            // right   `  `  (+z)
+            [0.0, 0.0, 1.0],
+            [0.0, 1.0, 1.0],
+            [1.0, 1.0, 1.0],
+            [1.0, 0.0, 1.0],
+            // left    `  `  (-z)
+            [0.0, 0.0, 0.0],
+            [0.0, 1.0, 0.0],
+            [1.0, 1.0, 0.0],
+            [1.0, 0.0, 0.0],
+        ],
+    );
+
+    // Take a look at the custom image (assets/textures/custom_image_for_example.png)
+    // so the UV coords will make more sense (note (0.0, 0.0) = Top-Left in UV mapping).
+    #[rustfmt::skip]
+    cube_mesh.insert_attribute(
+        Mesh::ATTRIBUTE_UV_0,
+        vec![
+            // Assigning the UV coords for the top side.
+            [0.5, 0.0], [0.0, 0.0], [0.0, 0.5], [0.5, 0.5],
+            // Assigning the UV coords for the bottom side.
+            [0.5, 0.5], [0.5, 0.0], [1.0, 0.0], [1.0, 0.5],
+            // Assigning the UV coords for the forward side.
+            [0.5, 1.0], [0.5, 0.5], [1.0, 0.5], [1.0, 1.0],
+            // Assigning the UV coords for the back side (same as forward because they have the
+            // same texture)
+            [0.5, 1.0], [0.5, 0.5], [1.0, 0.5], [1.0, 1.0],
+            // Assigning the UV coords for the right side.
+            [0.0, 1.0], [0.0, 0.5], [0.5, 0.5], [0.5, 1.0],
+            // Assigning the UV coords for the left side (same as right).
+            [0.0, 1.0], [0.0, 0.5], [0.5, 0.5], [0.5, 1.0],
+        ],
+    );
+
+    // When it comes to smooth, and simple meshes, normals are as simple as the direction of the flat surface
+    // Assign normals to allow for correct lighting calculations.
+    #[rustfmt::skip]
+    cube_mesh.insert_attribute(
+        Mesh::ATTRIBUTE_NORMAL,
+        vec![
+            // Normals for the top side (towards +y)
+            [0.0, 1.0, 0.0],
+            [0.0, 1.0, 0.0],
+            [0.0, 1.0, 0.0],
+            [0.0, 1.0, 0.0],
+            // Normals for the bottom side (towards -y)
+            [0.0, -1.0, 0.0],
+            [0.0, -1.0, 0.0],
+            [0.0, -1.0, 0.0],
+            [0.0, -1.0, 0.0],
+            // Normals for the forward side (towards +x)
+            [1.0, 0.0, 0.0],
+            [1.0, 0.0, 0.0],
+            [1.0, 0.0, 0.0],
+            [1.0, 0.0, 0.0],
+            // Normals for the back side (towards -x)
+            [-1.0, 0.0, 0.0],
+            [-1.0, 0.0, 0.0],
+            [-1.0, 0.0, 0.0],
+            [-1.0, 0.0, 0.0],
+            // Normals for the right side (towards +z)
+            [0.0, 0.0, 1.0],
+            [0.0, 0.0, 1.0],
+            [0.0, 0.0, 1.0],
+            [0.0, 0.0, 1.0],
+            // Normals for the left side (towards -z)
+            [0.0, 0.0, -1.0],
+            [0.0, 0.0, -1.0],
+            [0.0, 0.0, -1.0],
+            [0.0, 0.0, -1.0],
```

Comment:
Hmm, what's going on here? Checking out the direction methods on `Transform` I see they do not correspond at all to the directions presented here: <https://docs.rs/bevy/latest/bevy/prelude/struct.Transform.html#method.forward> (forward is -z)